src/PlusDataCollection/OpenHaptics/vtkPlusOpenHapticsDevice.cxx

```diff
@@ -0,0 +1,297 @@
+  trackerConfig->SetAttribute("DeviceName", this->DeviceName.c_str());
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalStartRecording()
+{
+  if (!this->isHapticDeviceInitialized)
+  {
+    LOG_ERROR("InternalStartRecording failed: OpenHaptics device has not been initialized");
+    return PLUS_FAIL;
+  }
+  LOG_TRACE("vtkPlusOpenHapticsDevice::InternalStartRecording");
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalStopRecording()
+{
+  LOG_TRACE("vtkPlusOpenHapticsDevice::InternalStopRecording");
+  return PLUS_SUCCESS;
+}
+
+HDCallbackCode HDCALLBACK
+vtkPlusOpenHapticsDevice::positionCallback(void* pData)
+{
+  
+  vtkPlusOpenHapticsDevice* client = reinterpret_cast<vtkPlusOpenHapticsDevice*>(pData);
+  HHD handle = client->DeviceHandle;
+  HD_state state = client->DeviceState;
+
+  const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
+  ++client->FrameNumber;
+
+  HDdouble force[3];
+  force[0] = 0;
+  force[1] = 0;
+  force[2] = 0;
+
+  //Transform to RAS coordinate system
+  vtkSmartPointer<vtkMatrix4x4> rasCorrection = vtkSmartPointer<vtkMatrix4x4>::New();
+  rasCorrection->Identity();
+  rasCorrection->SetElement(0, 0, -1);
+  rasCorrection->SetElement(1, 1, 0);
+  rasCorrection->SetElement(1, 2, 1);
+  rasCorrection->SetElement(2, 2, 0);
+  rasCorrection->SetElement(2, 1, 1);
+
+  //assemble force data
+  auto inputChannels = client->InputChannels;
+  vtkPlusDataSource* forceInput;
+  if (!inputChannels.empty() && inputChannels[0]->GetOwnerDevice()->GetToolByPortName("Force", forceInput))
+  {
+    StreamBufferItem* item = new StreamBufferItem();
+    if (forceInput->GetLatestItemUidInBuffer() > 0)
+    {
+      forceInput->GetLatestStreamBufferItem(item);
+      vtkSmartPointer<vtkMatrix4x4> forceMatrix = vtkSmartPointer<vtkMatrix4x4>::New();
+      item->GetMatrix(forceMatrix);
+      vtkMatrix4x4::Multiply4x4(rasCorrection, forceMatrix, forceMatrix);
+      force[0] = forceMatrix->GetElement(0,3);
+      force[1] = forceMatrix->GetElement(1,3);
+      force[2] = forceMatrix->GetElement(2,3);
+    }
+    delete item;
+  }
+  else
+  {
+    LOG_TRACE("No force data available");
+  }
+
+  //Current position data is pulled here
+  HDdouble trans[16];
+  HDdouble pos[3];  
+  HDdouble vel[3];
+  HDint buttonVals = 0;
+  HDboolean inkwell = 0;
+  hdBeginFrame(handle);
+  hdMakeCurrentDevice(handle);  
+  hdGetDoublev(HD_CURRENT_TRANSFORM, trans);
+  hdGetDoublev(HD_CURRENT_POSITION, pos);
+  hdGetDoublev(HD_CURRENT_VELOCITY, vel);
+  hdSetDoublev(HD_CURRENT_FORCE, force);
+  hdGetIntegerv(HD_CURRENT_BUTTONS, &buttonVals);
+  hdGetBooleanv(HD_CURRENT_INKWELL_SWITCH, &inkwell);
+  hdEndFrame(handle);
+
+  vtkPlusDataSource* stylus = NULL;
+  vtkPlusDataSource* velocity = NULL;
+  vtkPlusDataSource* buttons = NULL;
+
+  vtkSmartPointer<vtkTransform> transform = vtkSmartPointer<vtkTransform>::New();

```

Comment:
If these are allocated and deallocated every update, might be worth it to make them member variables to optimize.